frontend/catalyst/jax_primitives.py

```diff
@@ -684,8 +693,8 @@ def _grad_lowering(ctx, *args, jaxpr, fn, grad_params):
     new_argnums = [num + offset for num in argnums]
     argnum_numpy = np.array(new_argnums)
     diffArgIndices = ir.DenseIntElementsAttr.get(argnum_numpy)
-
-    _func_lowering(ctx, *args, call_jaxpr=jaxpr.eqns[0].params["call_jaxpr"], fn=fn, call=False)
+    func_call_jaxpr = _get_call_jaxpr(jaxpr)
+    _func_lowering(ctx, *args, call_jaxpr=func_call_jaxpr, fn=fn, call=False)
```

Comment:
Was this also broken for the grad lowering or did you just to update it pro-actively? :)

Reply:
Ah good to know. Could you add those tests and mark them xfail (or skip if they crash python)? :)

Reply:
I added some tests that are failing right now.